### PR TITLE
[Consorsbank] Add new format for documents newer than dec 2020

### LINKED ISSUE #310: [Consorsbank] Import throws JavaScript error in big.js
Uncaught (in promise) Error: [big.js] Invalid number
    at a6a55cb4657c6039bf55.js:1
    at new t (a6a55cb4657c6039bf55.js:1)
    at t (a6a55cb4657c6039bf55.js:1)
    at Gn (a6a55cb4657c6039bf55.js:1)
    at Module.Hn (a6a55cb4657c6039bf55.js:1)
    at li (a6a55cb4657c6039bf55.js:1)
    at a6a55cb4657c6039bf55.js:1
    at l (a6a55cb4657c6039bf55.js:1)
    at Generator.t._invoke (a6a55cb4657c6039bf55.js:1)
    at Generator.forEach.e.<computed> [as next] (a6a55cb4657c6039bf55.js:1)

## Documents
<details><summary>KAUF_898864719_ord176152884_001_wknDWS2W9_dat20200916_id908421543.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Fvwtie Zfyu",
    "Gdnpkjctf. 75",
    "78533 Lvwcqlzgx",
    "Consorsbank • 90318 Nürnberg",
    "ORDERABRECHNUNG",
    "KAUF",
    "AM 16.09.2020",
    "UM",
    "08:24:08",
    "AUSL.INVESTM.GESCH.",
    "YV. 241920098.506",
    "Depotnummer",
    "2470529332",
    "6734650071/05",
    "Vermerk der Bank",
    "1000",
    "02",
    "Hinweis für Orderabrechnungen und Depotbuchungsanzeigen:",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der",
    "Allgemeinen Geschäftsbedingungen (AGB Banken).",
    "Machen Sie Ihre Einwendungen in Textform",
    "geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank oder per Fax oder Mail an die",
    "unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Bezeichnung",
    "WKN",
    "ISIN",
    "DWS IN.-AR.INT. LCEOA",
    "DWS2W9",
    "LU1863263346",
    "Einheit",
    "Umsatz",
    "ST",
    "1,74447",
    "Preis pro Anteil",
    "143,310000",
    "EUR",
    "Nettoinventarwert",
    "EUR",
    "250,00",
    "Wert 21.09.2020",
    "EUR",
    "250,00",
    "Consorsbank",
    "Ausgabegebühr 0,00%",
    "sglfuxpv Aiubh-Vt. 0996368786",
    "Bestand zugunsten Girosammelverwahrung",
    "CBF 2126",
    "Ausgabeaufschlag regulär 5,00%",
    "Sparplanname: DWS Invest AI",
    "Beratungsfreies Geschäft"
  ]
]

```

</p></details>

<details><summary>KAUF_898864719_ord178558665_001_wknDWS2W9_dat20201016_id919254969.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Ijmwhu Qzfb",
    "Rmiebqdiq. 49",
    "25445 Zebsuoecr",
    "Consorsbank • 90318 Nürnberg",
    "ORDERABRECHNUNG",
    "KAUF",
    "AM 16.10.2020",
    "UM",
    "08:24:16",
    "AUSL.INVESTM.GESCH.",
    "ED. 806282561.155",
    "Depotnummer",
    "0271340215",
    "9519373321/77",
    "Vermerk der Bank",
    "1000",
    "02",
    "Hinweis für Orderabrechnungen und Depotbuchungsanzeigen:",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der",
    "Allgemeinen Geschäftsbedingungen (AGB Banken).",
    "Machen Sie Ihre Einwendungen in Textform",
    "geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank oder per Fax oder Mail an die",
    "unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Bezeichnung",
    "WKN",
    "ISIN",
    "DWS IN.-AR.INT. LCEOA",
    "DWS2W9",
    "LU1863263346",
    "Einheit",
    "Umsatz",
    "ST",
    "1,58306",
    "Preis pro Anteil",
    "154,070000",
    "EUR",
    "Nettoinventarwert",
    "EUR",
    "243,90",
    "Consorsbank",
    "Ausgabegebühr 2,50%",
    "EUR",
    "6,10",
    "Wert 21.10.2020",
    "EUR",
    "250,00",
    "ubvtntlu Tcgfq-Co. 5968267226",
    "Bestand zugunsten Girosammelverwahrung",
    "CBF 2126",
    "Ausgabeaufschlag regulär 5,00%",
    "Sparplanname: DWS Invest AI",
    "Beratungsfreies Geschäft"
  ]
]

```

</p></details>

<details><summary>KAUF_898864719_ord181538015_001_wknDWS2W9_dat20201117_id926579317.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "Vnafax Ffpf",
    "Xnxxiywom. 40",
    "23889 Doffrmaaw",
    "Consorsbank • 90318 Nürnberg",
    "ORDERABRECHNUNG",
    "KAUF",
    "AM 17.11.2020",
    "UM",
    "10:51:08",
    "AUSL.INVESTM.GESCH.",
    "KN. 696577651.654",
    "Depotnummer",
    "1667758507",
    "1567302027/54",
    "Vermerk der Bank",
    "1000",
    "02",
    "Hinweis für Orderabrechnungen und Depotbuchungsanzeigen:",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der",
    "Allgemeinen Geschäftsbedingungen (AGB Banken).",
    "Machen Sie Ihre Einwendungen in Textform",
    "geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank oder per Fax oder Mail an die",
    "unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Bezeichnung",
    "WKN",
    "ISIN",
    "DWS IN.-AR.INT. LCEOA",
    "DWS2W9",
    "LU1863263346",
    "Einheit",
    "Umsatz",
    "ST",
    "1,60009",
    "Preis pro Anteil",
    "152,430000",
    "EUR",
    "Nettoinventarwert",
    "EUR",
    "243,90",
    "Consorsbank",
    "Ausgabegebühr 2,50%",
    "EUR",
    "6,10",
    "Wert 20.11.2020",
    "EUR",
    "250,00",
    "lvkfzpzy Oeyah-Nf. 9814174754",
    "Bestand zugunsten Girosammelverwahrung",
    "CBF 2126",
    "Ausgabeaufschlag regulär 5,00%",
    "Sparplanname: DWS Invest AI",
    "Beratungsfreies Geschäft"
  ]
]

```

</p></details>

## Community profile

https://community.tresor.one/u/ad0e19fb

---
Issue submitted with https://import.tresor.one

### LINKED ISSUE #462: [consorsbank] Number can't be parsed
## Documents
<details><summary>document0.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "HRZZFIWMO YkxU",
    "Mohhvr Ika 65",
    "13811 Omivjnrg",
    "Consorsbank • 90318 Nürnberg",
    "ORDERABRECHNUNG",
    "KAUF",
    "AM",
    "09.12.2020",
    "UM",
    "21:18:29",
    "TRADEGATE EXCHANGE",
    "AD. 554894853.577",
    "Depotnummer:",
    "1978 755 200",
    "6022229583/76",
    "Vermerk der Bank:",
    "7514",
    "9",
    "Datum:",
    "10.12.2020",
    "Seite:",
    "1 von 1",
    "Ordernummer",
    "058104755.625",
    "Orderdatum",
    "09.12.2020",
    "Hinweis für Orderabrechnungen und Depotbuchungsanzeigen:",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der",
    "Allgemeinen Geschäftsbedingungen (AGB Banken).",
    "Machen Sie Ihre Einwendungen in Textform",
    "geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank oder per Fax oder Mail an die",
    "unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Bezeichnung",
    "WKN",
    "ISIN",
    "ADOBE INC.",
    "871981",
    "US00724F1012",
    "Einheit",
    "Umsatz",
    "ST",
    "7,00000",
    "Preis pro Anteil",
    "400,000000 EUR",
    "FRANCO",
    "COURTAGE",
    "Kurswert",
    "2.800,00 EUR",
    "Provision",
    "7,00 EUR",
    "Grundgebühr",
    "4,95 EUR",
    "zulasten Konto-Nr. 1234567890",
    "2.811,95 EUR",
    "Bestand zugunsten Girosammelverwahrung",
    "CBF 2126",
    "Limitkurs  400,000000 EUR",
    "Beratungsfreies Geschäft"
  ]
]

```

</p></details>

<details><summary>document1.pdf</summary><p>

```json
[
  [
    "Consorsbank ist eine eingetragene Marke der BNP Paribas S.A. Niederlassung Deutschland (AG nach franz. Recht).",
    "Standort Nürnberg: Bahnhofstraße 55, 90402 Nürnberg, HRB Nürnberg 31129, USt-IdNr. DE191528929",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00, info@consorsbank.de, www.consorsbank.de",
    "Sitz der BNP Paribas S.A.: 16, boulevard des Italiens, 75009 Paris, Frankreich, Registergericht: R.C.S. Paris 662 042 449",
    "Président du Conseil d‘Administration (Präsident des Verwaltungsrates): Jean Lemierre, Directeur Général (Generaldirektor): Jean-Laurent Bonnafé",
    "SDLINMTXR QucX",
    "Tcajit You 55",
    "60840 Ihuncsmh",
    "Consorsbank • 90318 Nürnberg",
    "ORDERABRECHNUNG",
    "KAUF",
    "AM",
    "09.12.2020",
    "UM",
    "19:42:10",
    "TRADEGATE EXCHANGE",
    "UH. 284499964.962",
    "Depotnummer:",
    "5091 028 342",
    "9404225314/94",
    "Vermerk der Bank:",
    "6724",
    "9",
    "Datum:",
    "10.12.2020",
    "Seite:",
    "1 von 1",
    "Ordernummer",
    "176591792.019",
    "Orderdatum",
    "09.12.2020",
    "Hinweis für Orderabrechnungen und Depotbuchungsanzeigen:",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der",
    "Allgemeinen Geschäftsbedingungen (AGB Banken).",
    "Machen Sie Ihre Einwendungen in Textform",
    "geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an die Consorsbank oder per Fax oder Mail an die",
    "unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Bezeichnung",
    "WKN",
    "ISIN",
    "FREEPORT-MCMORAN INC.",
    "896476",
    "US35671D8570",
    "Einheit",
    "Umsatz",
    "ST",
    "125,00000",
    "Preis pro Anteil",
    "20,000000 EUR",
    "FRANCO",
    "COURTAGE",
    "Kurswert",
    "2.500,00 EUR",
    "Provision",
    "6,25 EUR",
    "Grundgebühr",
    "4,95 EUR",
    "zulasten Konto-Nr. 1234567890",
    "2.511,20 EUR",
    "Bestand zugunsten Girosammelverwahrung",
    "CBF 2126",
    "Limitkurs  20,000000 EUR",
    "Beratungsfreies Geschäft"
  ]
]

```

</p></details>

## Community profile

Reported by user, @dredav

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by dredav**
- Fix #462
- Add test case from #462
- I've don't change the tax logic because of missing test files
- Fix #310 (Add support for funds)
- Add test case from #310

### CODE REVIEW

File: src/brokers/consorsbank.js [hunk 1]
```diff
@@ -134,21 +163,37 @@ const findAmount = (textArr, type) => {
...
-    Big(parseGermanNum(baseFee))
+const getNumberAfterTermWithOffset = (content, termToLower, offset = 0) => {
+  const lineNumber = content.findIndex(line =>
+    line.toLowerCase().includes(termToLower)
   );
 
-  return Math.abs(sum);
+  if (lineNumber <= 0) {
+    return 0;
+  }
```
Comment (SirGibihm) [inline src/brokers/consorsbank.js:178]: Wouldn't it be better to return `undefined` or at least `-1`?  When handled like this the case where 0 is the right number can not be differentiated from a mistake or am I seing this wrong?

File: src/brokers/consorsbank.js [hunk 2]
```diff
@@ -134,21 +163,37 @@ const findAmount = (textArr, type) => {
...
+
+  if (/^[A-Z]{3}$/.test(content[lineNumber + offset + 1])) {
+    // Documents before dec 2020 have the price after the currency line.
+    return parseGermanNum(content[lineNumber + offset + 2]);
+  }
```
Comment (SirGibihm) [inline src/brokers/consorsbank.js:183]: Are you testing on strings such as `EUR` or `USD` here? If so, are all currencies always three letters? Could it be more or less?

File: src/brokers/consorsbank.js [hunk 3]
```diff
@@ -134,21 +163,37 @@ const findAmount = (textArr, type) => {
...
+
+const findFee = content => {
+  const feeBrokerage = getNumberAfterTermWithOffset(content, 'provision');
+  const feeBase = getNumberAfterTermWithOffset(content, 'grundgebühr');
+  let feeIssue = 0;
+  if (content.indexOf('Ausgabegebühr 0,00%') <= 0) {
+    feeIssue = getNumberAfterTermWithOffset(content, 'ausgabegebühr');
+  }
+
+  return Math.abs(+Big(feeBrokerage).plus(Big(feeBase)).plus(Big(feeIssue)));
```
Comment (SirGibihm) [inline src/brokers/consorsbank.js:196]: Why use `Math.abs`? Could the `Ausgabegebühr` be a negative percentage? In that case we would have a discount which we saw in other brokers.

**Review by SirGibihm — COMMENTED** [review]
Minute questions, if there is a solid answer I see no issues in merging this PR!

File: src/brokers/consorsbank.js (hunk 2 above)
Comment (dredav) [reply src/brokers/consorsbank.js:183]: I'm not sure if all brokers are using the `ISO 4217 currency codes`, but the most did. So we also currently need the three letter code for the `foreignCurrency` [field](https://github.com/codeofsumit/tresor-import/blob/master/docs/activity.md).

File: src/brokers/consorsbank.js (hunk 3 above)
Comment (dredav) [reply src/brokers/consorsbank.js:196]: Your right, I'll remove the `Math.abs` call.

File: src/brokers/consorsbank.js (hunk 1 above)
Comment (dredav) [reply src/brokers/consorsbank.js:178]: That something relevant is in line 0 is for me also not a normal case. But your point is easier to understand, I'll change that.

**Comment by SirGibihm** [comment]
Just retag me once the changes are in and I will merge

**Comment by dredav** [comment]
Changes pushed @SirGibihm!

**Review by SirGibihm — APPROVED** [review]
Nice one, another one down

**Pull request merged**